tests/integration/components/rive-animation-test.js

```diff
@@ -0,0 +1,176 @@
+    }
+  }
+}
+
+module('Integration | Component | rive-animation', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.originalRive = window.Rive;
+    window.Rive = class MockRiveConstructor {
+      constructor(options) {
+        const instance = new MockRive(options);
+        this._onLoadCallback = options.onLoad;
+        Object.assign(this, instance);
+      }
+    };
+  });
+
+  hooks.afterEach(function () {
+    window.Rive = this.originalRive;
+  });
+
+  test('it renders and initializes correctly', async function (assert) {
+    await render(hbs`
+      <div class="w-[200px] h-[200px]">
+        <RiveAnimation @src="/assets/animations/gleam_logo_animation.riv" />
+      </div>
+    `);
+
+    // Check container dimensions
+    const container = this.element.querySelector('div > div');
+    assert.ok(container, 'Container element exists');
+    // Check canvas dimensions and styles
+    const canvas = this.element.querySelector('canvas');
+    assert.ok(canvas, 'Canvas element exists');
+    assert.strictEqual(canvas?.className, 'w-full h-full block max-w-full', 'Canvas has correct classes');
+
+    // Create and attach mock Rive instance
+    const mockRive = new MockRive({
+      canvas: canvas,
+      src: '/assets/animations/gleam_logo_animation.riv',
+      layout: { fit: 'contain' },
+    });
+    container.__riveInstance = mockRive;
+
+    // Simulate Rive load event
+    mockRive.simulateLoad();
+
+    // Check that Rive instance was created with correct options
+    assert.ok(mockRive, 'Rive instance exists');
+    assert.strictEqual(mockRive.canvas, canvas, 'Canvas is passed to Rive instance');
+    assert.strictEqual(mockRive.src, '/assets/animations/gleam_logo_animation.riv', 'Correct animation source is set');
+    assert.ok(mockRive.layout, 'Layout is configured');
+  });
+
+  test('it handles hover state correctly', async function (assert) {
+    await render(hbs`
+      <div class="w-[200px] h-[200px]">
+        <RiveAnimation @src="/assets/animations/gleam_logo_animation.riv" />
+      </div>
+    `);
+
+    const container = this.element.querySelector('div > div');
+    assert.ok(container, 'Container exists');
+
+    // Create and attach mock Rive instance
+    const mockRive = new MockRive({ canvas: container.querySelector('canvas') });
+    container.__riveInstance = mockRive;
+    assert.ok(mockRive, 'Rive instance exists');
+
+    // Simulate Rive load event
+    mockRive.simulateLoad();
+
+    const inputs = mockRive.stateMachineInputs();
+    assert.ok(inputs, 'State machine inputs exist');
+
+    const hoverInput = inputs.find((input) => input.name.toLowerCase().includes('hover'));
+    assert.ok(hoverInput, 'Hover input exists');
+
+    // Verify initial hover state
+    assert.false(hoverInput.value, 'Hover input is initially false');
+
+    // Wait for hover out timeout
+    assert.false(hoverInput.value, 'Hover input is set to false after timeout');
```

Comment:
Where is the "timeout" here? Also seems like we're just testing the mocked static array above here, don't really see the point of the test